connectors/sources/mssql.py

```diff
@@ -419,55 +563,88 @@ async def fetch_documents(self, table):
             Dict: Document to be indexed
         """
         try:
-            row_count = await self.mssql_client.get_table_row_count(table=table)
-            if row_count > 0:
-                # Query to get the table's primary key
-                keys = await self.mssql_client.get_table_primary_key(table=table)
-                keys = map_column_names(
-                    column_names=keys, schema=self.schema, table=table
+            docs_generator = (
+                self._yield_docs_custom_query(tables, query)
+                if query is not None
+                else self._yield_all_docs_from_tables(tables[0])
+            )
+            async for doc in docs_generator:
+                yield doc
+        except (InternalClientError, ProgrammingError) as exception:
+            self._logger.warning(
+                f"Something went wrong while fetching document for table {', '.join(tables)}. Error: {exception}"
```

Comment:
Might want to log `query` as well. Or maybe make it two methods:
1. fetch_documents_from_table
2. fetch_documents_from_query

Reply:
Sure, I have created two methods